llvm/include/llvm/Analysis/DXILResource.h

```diff
@@ -216,15 +240,88 @@ class ResourceInfo {
 
   MDTuple *getAsMetadata(LLVMContext &Ctx) const;
 
-  ResourceBinding getBinding() const { return Binding; }
   std::pair<uint32_t, uint32_t> getAnnotateProps() const;
 
   void print(raw_ostream &OS) const;
 };
 
 } // namespace dxil
 
-using DXILResourceMap = MapVector<CallInst *, dxil::ResourceInfo>;
+class DXILResourceMap {
+  SmallVector<dxil::ResourceInfo> Resources;
+  DenseMap<CallInst *, unsigned> CallMap;
```

Comment:
What is the function called by these CallInst here and who is the user of these CallInst when use ResourceAnalysis?

Reply:
These are the `dx.handle.fromBinding` calls as you can see in the test. We could/should arguably update them to point at the `dx.op.createHandle*` calls when we replace those, but that's not currently implemented.

See https://github.com/llvm/llvm-project/pull/104251/files#diff-abc4769e41f4d9022e3b88d8d752267e5d1bb26c0d4c4186f5d551c0dce76b80R177 for an example of the usage.

Reply:
I see.  It is used in lowering right now.
If no other uses after the lowering, maybe it is OK to not update it?